packages/owl-bot/src/copy-code.ts

```diff
@@ -587,17 +631,54 @@ export async function copyExists(
     return false;
   };
 
-  if (findInBodies('Pull request', pulls)) return true;
+  for (let page = 1, prsSeen = 0; prsSeen < searchDepth; page += 1) {
+    const request: Request = {
+      owner,
+      repo,
+      per_page,
+      state: 'all',
+      page,
+    };
+    const cacheKey = keyFrom('pulls', request);
+    const cachedPulls = cache.get(cacheKey);
+    let pulls;
+    if (cachedPulls) {
+      pulls = cachedPulls;
+    } else {
+      pulls = await octokit.pulls.list({...request});
+      cache.set(cacheKey, pulls);
+    }
```

Comment:
nit: I might have designed this as a read-through cache (just ask the cache for page x of issues|pulls and it'll either return the cached version or fetch it, store it, and return it to you), rather than exposing it as a `Map` that you `get` and `set` on. Internally, it could do this same logic, but hide that from the calling code.

```
const cache = newRepoHistoryCache(octokit);
...
const pulls = await cache.fetchPulls(request);
```

Reply:
I expect the calling code to treat it as an opaque value, as it currently does.  Note, that I could completely change type RepoHistoryCache without changing any of the calling code.

Reply:
If RepoCacheHistory ever gets passed to another function, then I'll wrap it in a type as you described.